homeassistant/components/switch/tplink.py

```diff
@@ -53,6 +53,18 @@ def __init__(self, smartplug, name):
         # Set up emeter cache
         self._emeter_params = {}
 
+        if self._emeter_present:
```

Comment:
Instead of this, just pass `True` as a second param to `add_devices`. That way HASS will call `update()` before setting initial state.